tools/airgap_linter.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+#
+# Validates that the supplied framework (identified by the directory passed in) will function, to the best
+# of our knowledge, in an airgapped cluster.
+#
+# The checks are:
+#   - No external URIs defined in anything but resource.json
+#   - No images defined in anything but resource.json (and templated into the svc.yml)
+#
+
+import re
+import sys
+import os
+
+
+def extract_uris(file_name):
+    with open(file_name, "r") as file:
+        lines = file.readlines()
+
+    matcher = re.compile(".*(https:\/\/|http:\/\/)([^\/]*)", re.IGNORECASE)
```

Comment:
I'm not sure I understand this regex.
It's looking for https:// or http:// followed by any number of non-slashes, and it is trying to match an entire line.

so, wouldn't something like
```
     my_url = "https://something.com/something/something"
```
be a non-match?  The entire line can't be matched, because although it contains https://, it has another slash after the base protocol identifier.

Oh, I guess .match only anchors at the start of the subject, not the end (what a weird interface?)

Still, you will get very odd results for things like

```
      my_url = "https://fruitloops.com" // This is a url that I love
      my_url2 = "https://somethingelse.com?param=arg&param2=arg2&param3&arg3"
```

I'd counter-suggest
```
    with open(file_name, "r") as source_f:
        url_matcher = re.compile(r"https?://([^/]+)/", re.IGNORECASE)
        matches = []
        for line in source_f: # iterates across readline
            for match in re.finditer(url_matcher, line):
                 matches.append(match.group(1))
        return matches
```

(aside, cpython implicitly compiles inline strings, but probably best to explicitly compile to avoid confusion)